app/Controllers/MediaController.php

```diff
@@ -188,6 +188,50 @@ public function download(Request $request, Response $response, string $userCode,
         return $this->streamMedia($request, $response, $this->storage, $media, 'attachment');
     }
 
+
+    /**
+     * @param  Request  $request
+     * @param  Response  $response
+     * @param  string  $vanity
+     * @param  string  $id
+     *
+     * @return Response
+     * @throws HttpNotFoundException
+     * @throws HttpBadRequestException
+     */
+    public function createVanity(Request $request, Response $response, int $id): Response
+    {
+        if (!$this->session->get('admin')) {
+            $media = $this->database->query('SELECT * FROM `uploads` WHERE `id` = ? LIMIT 1', $id)->fetch();
+        } else {
+            $media = $this->database->query(
+                'SELECT * FROM `uploads` WHERE `id` = ? AND `user_id` = ? LIMIT 1',
+                [$id, $this->session->get('user_id')]
+            )->fetch();
+        }
+
+        $data = $request->getParsedBody();
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏There is no request validation here

Reply:
Im a little confused at what you mean there is no request validation. If you mean checking if the media does exist, I already have that in my code:

https://github.com/SergiX44/XBackBone/pull/507/files#diff-1f64a59ac49e401b085fe29e9b0d34bb824ebd81e54c8416c2e0e2df674b14a3R217-R219